client/src/plugins/settings/__tests__/SettingsFormSpec.js

```diff
@@ -141,6 +144,290 @@ describe('SettingsForm', function() {
+              type: 'boolean',
+              label: 'Checkbox'
+            },
+          }
+        },
+        {
+          id:'test',
+          properties: {
+
+            'test.text': {
+              type: 'text',
+              label: 'Text Input',
+              condition: { property: 'otherSection.checkbox', equals: true }
+            }
+          }
+        }
+      ];
+
+      const { container } = createSettingsForm({
+        schema,
+        initialValues: {
+          'otherSection': {
+            'checkbox': false
+          }
+        }
+      });
+
+      // assert
+      let textField = container.querySelector('.form-group [id="test.text"]');
+      expect(textField).to.not.exist;
+
+      // when
+      fireEvent.click(container.querySelector('.form-group [id="otherSection.checkbox"]'));
+
+      // then
+      textField = container.querySelector('.form-group [id="test.text"]');
+      expect(textField).to.exist;
+    });
+
+
+    it('should show field if condition "equals string" is met', function() {
+
+      // given
+      const schema = [ {
+        properties: {
+          'test.text': {
+            type: 'text',
+            label: 'Text to check'
+          },
+          'test.conditionalText': {
+            type: 'text',
+            label: 'shown if text is "show"',
+            condition: { property: 'text', equals: 'show' }
+          }
+        }
+      } ];
+
+      const { container } = createSettingsForm({
+        schema,
+        initialValues: {
+          'test': {
+            'text': ''
+          }
+        }
+      });
+
+      // assert
+      let conditionalText = container.querySelector('.form-group [id="test.conditionalText"]');
+      expect(conditionalText).to.not.exist;
+
+      // when
+      fireEvent.change(container.querySelector('.form-group [id="test.text"]'), { target: { value: 'show' } });
+
+      // then
+      conditionalText = container.querySelector('.form-group [id="test.conditionalText"]');
+      expect(conditionalText).to.exist;
+    });
+
+
+    it('should show field if condition "oneOf" is met', function() {
+
+      // given
+      const schema = [ {
+        properties: {
+          'test.select': {
+            type: 'select',
+            label: 'Environment',
+            options: [
+              { label: 'Development', value: 'dev' },
+              { label: 'Staging', value: 'staging' },
+              { label: 'Production', value: 'prod' }
+            ]
+          },
+          'test.conditionalText': {
+            type: 'text',
+            label: 'shown if select is "dev" or "staging"',
+            condition: {
+              property: 'select',
+              oneOf: [ 'dev', 'staging' ]
+            }
+          }
+        }
+      } ];
+
+      const { container } = createSettingsForm({
+        schema,
+        initialValues: {
+          'test': {
+            'select': 'prod'
+          }
+        }
+      });
+
+      // assert
```

Comment:
I usually call this `// assume` so it is clear it verifies the test setup, before the actual test.